apps/src/lib/tools/jsdebugger/JsDebugger.jsx

```diff
@@ -205,7 +233,119 @@ var UnconnectedJsDebugger = Radium(React.createClass({
     this.setState({transitionType: null});
   },
 
+  onMouseDownDebugResizeBar(event) {
+    // When we see a mouse down in the resize bar, start tracking mouse moves:
+    var eventSourceElm = event.srcElement || event.target;
+    if (eventSourceElm.id === 'debugResizeBar') {
+      this._draggingDebugResizeBar = true;
+      document.body.addEventListener('mousemove', this.onMouseMoveDebugResizeBar);
+      const mouseMoveTouchEventName = dom.getTouchEventName('mousemove');
+      if (mouseMoveTouchEventName) {
+        document.body.addEventListener(
+          mouseMoveTouchEventName,
+          this.onMouseMoveDebugResizeBar
+        );
+      }
+
+      event.preventDefault();
+    }
+  },
+
+  /**
+   *  Handle mouse moves while dragging the debug resize bar.
+   */
+  onMouseMoveDebugResizeBar(event) {
+    var codeApp = document.getElementById('codeApp');
+    var codeTextbox = document.getElementById('codeTextbox');
```

Comment:
I still don't love the idea of this component reaching outside to modify the dimensions of these other DOM elements - I'd rather have something a layer or two up in charge of this.  Nonblocking to this refactor, but it'd be nice to address in the future.